File: back/app/models/calendar_event.py
# app/models/calendar_event.py
from sqlalchemy import Column, Integer, String, Date, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from app.core.database import Base

class CalendarEvent(Base):
    __tablename__ = "calendar_events"

    id = Column(Integer, primary_key=True)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)

    title = Column(String(255), nullable=False)
    start_date = Column(Date, nullable=False)
    end_date = Column(Date, nullable=True)

    level = Column(String(20), default="primary")  # success, danger, warning
    type = Column(String(20), default="custom")    # custom | loan

File: front/src/app/interfaces/calendar-event.ts
// calendar-event.model.ts
export type CalendarLevel =
  | 'primary'
  | 'success'
  | 'warning'
  | 'danger';

export type CalendarEventType =
  | 'custom'
  | 'loan';

export interface CalendarEvent {
  id: number;
  title: string;

  start: string;   // ISO date
  end?: string;

  level: CalendarLevel;
  type: CalendarEventType;
}
